### PR TITLE
chore: Force license uuids to not be null

### PR DESCRIPTION
**Pull request by Emyrk**
# What this does

All new licenses have uuids, old licenses might not. So this PR gives old licenses a random uuid to support a non-null database schema.

# Future work

We should require licenses to have uuids or fail. Breaking old licenses.

### CODE REVIEW

**Comment by Emyrk** [comment]
This did make unit tests fail: https://github.com/coder/coder/actions/runs/4080226583/jobs/7032466456#step:9:169

**Comment by spikecurtis** [comment]
It's unclear to me why we needed a UUID in the first place.

We had just an `id` index --- is it just to normalize with other DB objects where we use UUIDs for them?

**Comment by Emyrk** [comment]
@spikecurtis I am not sure. I do not think this PR is the fix, but something should be done.

**Comment by spikecurtis** [comment]
Looks like UUIDs were added in https://github.com/coder/coder/pull/5110 

@kylecarbs can you bring us up to speed on why UUIDs were added and whether we can drop one or other of `id` `uuid`?

**Comment by Emyrk** [comment]
We could make the code accept invalid uuids if the license is generated before Feb 8, 2023 for example. Any new licenses have to have valid uuids.

**Comment by Kira-Pilot** [comment]
@kylecarbs bump
This is blocking #6125. I would prefer it if we could stick with the UUID because the Audit Log [needs that type](https://github.com/coder/coder/blob/main/coderd/audit/request.go#L79) and seems like every other model has it.

**Comment by spikecurtis** [comment]
@Emyrk Is it now the case that the UUID is embedded in the signed license itself, or its it just a DB identifier local to the deployment it is installed on?

**Comment by Emyrk** [comment]
It is embedded to the license claims. We insert a uuid string in the JWT ID field: https://github.com/coder/license/blob/2ae8cabd512f05b58ab871d41e6327bbc405a00d/cmd/licensor/issue.go#L79-L79

**Comment by Emyrk** [comment]
@spikecurtis uuids were added to licenses 4 months ago: https://github.com/coder/license/commit/2d26cbd6f689b566a2e84b47d98f94748e2c7fa5

I am not sure if licenses exist without uuids. It is possible. However I think we could just force the error if the license is made after a certain date to handle any old formats.

**Comment by kylecarbs** [comment]
Would this stop old licenses from working @Emyrk?

**Comment by Emyrk** [comment]
@kylecarbs To support old licenses that fail `uuid` parse, I think we would need to check the issue date (if it's in there) and allow uuid to fail for licenses before a certain date.

This PR does not do that yet.

**Comment by Emyrk** [comment]
I am going to generate a random uuid for licenses that do not have uuids.

This will make old and new licenses work. A followup should be to **require** uuids, potentially breaking old licenses.

File: coderd/database/migrations/000097_license_not_null_uuid.up.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-- We need to assign uuids to any existing licenses that don't have them.
+UPDATE licenses SET uuid = gen_random_uuid() WHERE uuid IS NULL;
+-- Assert no licenses have null uuids.
+ALTER TABLE ONLY licenses ALTER COLUMN uuid SET NOT NULL;
```
Comment (coadler) [inline coderd/database/migrations/000097_license_not_null_uuid.up.sql:4]: should have a `BEGIN`/`COMMIT`.

**Review by coadler — APPROVED** [review]

**Pull request merged**